Composer/packages/client/src/pages/publish/BotStatusList.tsx

```diff
@@ -0,0 +1,325 @@
+
+import { IStatus, PublishStatusList } from './PublishStatusList';
+import { detailList, listRoot, tableView } from './styles';
+
+// Licensed under the MIT License.
+export type IBotStatus = {
+  id: string;
+  name: string;
+  publishTargets?: PublishTarget[];
+  publishTarget?: string;
+  time?: string;
+  status?: number;
+  message?: string;
+  comment?: string;
+};
+export type IBotStatusListProps = {
+  projectId: string;
+  items: IBotStatus[];
+  botPublishHistoryList: { projectId: string; publishHistory: IStatus[] }[];
+  updateItems: (items: IBotStatus[]) => void;
+  updatePublishHistory: (items: IStatus[], item: IBotStatus) => void;
+  updateSelectedBots: (items: IBotStatus[]) => void;
+  changePublishTarget: (PublishTarget: string, item: IBotStatus) => void;
+  onLogClick: (item: IStatus) => void;
+  onRollbackClick: (selectedVersion: IStatus, item: IBotStatus) => void;
+};
+
+export const BotStatusList: React.FC<IBotStatusListProps> = (props) => {
+  const {
+    projectId,
+    items,
+    botPublishHistoryList,
+    updateItems,
+    updatePublishHistory,
+    changePublishTarget,
+    updateSelectedBots,
+    onLogClick,
+    onRollbackClick,
+  } = props;
+  const [selectedBots, setSelectedBots] = useState<IBotStatus[]>([]);
+  const [showHistoryBots, setShowHistoryBots] = useState<string[]>([]);
+
+  const [currentSort, setSort] = useState({ key: 'Bot', descending: true });
+  const sortByName = (ev: React.MouseEvent<HTMLElement>, column: IColumn): void => {
+    if (column.isSorted) {
+      column.isSortedDescending = !column.isSortedDescending;
+      const newItems: IBotStatus[] = items.reverse();
+      updateItems(newItems);
+    }
+  };
+  const changeSelected = (item: IBotStatus, isChecked?: boolean) => {
+    let newSelectedBots: IBotStatus[];
+    if (isChecked) {
+      newSelectedBots = selectedBots.concat(item);
+    } else {
+      newSelectedBots = selectedBots.filter((bot) => bot.id !== item.id);
+    }
+    setSelectedBots(newSelectedBots);
+    updateSelectedBots(newSelectedBots);
+  };
+
+  const publishTargetOptions = (item: IBotStatus): IDropdownOption[] => {
+    const options: IDropdownOption[] = [];
+    item.publishTargets &&
+      item.publishTargets.forEach((target) => {
+        options.push({
+          key: target.name,
+          text: target.name,
+        });
+      });
+    options.push({
+      key: 'manageProfiles',
+      text: formatMessage('Manage profiles'),
+      data: { style: { color: '#0078D4' } },
+    });
+    return options;
+  };
+  const onRenderOption = (option?: IDropdownOption): JSX.Element | null => {
+    if (!option) return null;
+    return <div style={option.data && option.data.style}>{option.text}</div>;
+  };
+  const onRenderStatus = (item: IBotStatus): JSX.Element | null => {
+    if (!item.status) {
+      return null;
+    } else if (item.status === 200) {
+      return <Icon iconName="Accept" style={{ color: 'green', fontWeight: 600 }} />;
```

Comment:
lets not use green as a color. Also looking at the style for it this is not exactly green. Use SharedColors or NeutralColors.